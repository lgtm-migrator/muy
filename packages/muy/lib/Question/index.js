import React from "react"
import Box from "@material-ui/core/Box"
import Divider from "@material-ui/core/Divider"
import QuestionText from "../QuestionText"
import AnswerWithText from "../AnswerWithText"

const Question = ({
  children,
  question,
  AnswerProps,
  QuestionProps,
  ...props
}) => (
  <Box
    itemScope
    itemProp="mainEntity"
    itemType="https://schema.org/Question"
    {...props}
  >
    <QuestionText {...QuestionProps}>{question}</QuestionText>
    <AnswerWithText {...AnswerProps}>{children}</AnswerWithText>
    <Divider />
  </Box>
)

export default Question
