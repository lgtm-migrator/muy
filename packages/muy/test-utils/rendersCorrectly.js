import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
