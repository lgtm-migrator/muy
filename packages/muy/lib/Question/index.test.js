import React from "react"
import rendersCorrectly from "../../test-utils/rendersCorrectly"
import Question from "."

describe("<Question /> component", () => {
  it("renders correctly", () => {
    rendersCorrectly(<Question />)
  })
})
