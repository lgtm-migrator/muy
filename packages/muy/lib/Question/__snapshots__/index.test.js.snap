// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Question /> component renders correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-1"
  itemProp="mainEntity"
  itemScope={true}
  itemType="https://schema.org/Question"
>
  <dt
    className="MuiBox-root MuiBox-root-2"
  />
  <dd
    className="MuiBox-root MuiBox-root-3"
    itemProp="acceptedAnswer"
    itemScope={true}
    itemType="https://schema.org/Answer"
  >
    <div
      className="MuiBox-root MuiBox-root-4"
      itemProp="text"
    />
  </dd>
  <hr
    className="MuiDivider-root"
  />
</div>
`;
